<launch>

    <arg name="namespace" default="sam"/>
	<arg name="emergency_config_file" default="$(find sam_action_servers)/config/emergency_config.yaml"/>

    <group ns="$(arg namespace)">
    	<group ns="ctrl">
			<!-- Emergency surface action -->
			<node name="emergency_manager_node" pkg="sam_action_servers" type="emergency_manager_node" output="screen">
				<rosparam file="$(arg emergency_config_file)" command="load"/>
				<param name="emergency_config_file" value="$(arg emergency_config_file)"/>
				<param name="emergency_as" value="/$(arg namespace)/ctrl/emergency_surface_action" />
				<param name="abort_top" value="/$(arg namespace)/core/abort" />
				<param name="thruster_1_top" value="/$(arg namespace)/core/thruster1_cmd" />
				<param name="thruster_2_top" value="/$(arg namespace)/core/thruster2_cmd" />
				<param name="vbs_cmd_top" value="/$(arg namespace)/core/vbs_cmd" />
				<param name="goto_wp_action" value="/$(arg namespace)/ctrl/goto_waypoint" />
			</node>

		<node name="goto_waypoint" pkg="sam_action_servers" type="wp_goto_action.py" output="screen">
			<param name="base_frame" value="$$(arg namespace)/base_link"/>
			<param name="base_frame_2d" value="$(arg namespace)/base_link_2d"/>
			<param name="wp_tolerance" value="2."/>
			<param name="node_freq" value="40." />
			<!-- <param name="depth_tolerance" value="0.5"/> -->
			<param name="rpm1_cmd_topic" value="/$(arg namespace)/core/thruster1_cmd" />
			<param name="rpm2_cmd_topic" value="/$(arg namespace)/core/thruster2_cmd" />
			<param name="heading_setpoint_topic" value="/$(arg namespace)/ctrl/heading_pid/setpoint" />
			<param name="heading_state_topic" value="/$(arg namespace)/ctrl/heading_pid/state" />
			<param name="heading_enable_topic" value="/$(arg namespace)/ctrl/heading_pid/pid_enable" />
			<param name="elevator_setpoint_topic" value="/$(arg namespace)/ctrl/elevator_pid/setpoint" />
			<param name="elevator_enable_topic" value="/$(arg namespace)/ctrl/elevator_pid/pid_enable" />
			<param name="vbs_setpoint_topic" value="/$(arg namespace)/ctrl/vbs/setpoint" />
			<param name="rudder_angle" value="0.1" />
		</node>
	</group>

	</group>


	<!-- Planned surface node -->
	<!-- <include file="$(find sam_action_servers)/launch/mission_complete_node.launch" >
		<arg name="robot_name" value="$(arg namespace)"/>	
	</include> -->

    
</launch>
